src/SoftekBarcodeAPI.php

```diff
@@ -37,8 +40,101 @@ public function __construct(SoftekBarcodeConfig $configuration)
+     */
+    public function processImage(string $pathToFile): array
+    {
+        $result = $this->ffi->mtScanBarCode($this->instance, $pathToFile);
+        if ($result < 0) {
+            throw new SoftekProcessImageException($result, $pathToFile);
+        }
+
+        // if no barcodes detected return empty array
+        if ($result <= 0) return [];
+
+        return $this->extractScanResults($result);
+    }
+
+
+    /**
+     * This method extract the information using the barcode toolkit. It is required to run FFI::mtScanBarcode before
+     * this method to perform a legit request to the barcode toolkit interface.
+     *
+     * @param int $resultCount
+     * @return BarcodeScanResult[]
+     */
+    private function extractScanResults(int $resultCount): array
+    {
+        $barcodeScanResults = [];
+        for ($barcodeIndex = 1; $barcodeIndex <= $resultCount; $barcodeIndex++) {
+            $barcodeScanResult = new BarcodeScanResult();
+            // converting the cdate byte array type to a readable string
+            $barcodeScanResult
+                ->setText($this->ffiByteArrayToString($this->ffi->mtGetBarString($this->instance, $barcodeIndex)))
+                ->setType($this->ffiByteArrayToString($this->ffi->mtGetBarStringType($this->instance, $barcodeIndex)));
+
+            // switch type of variable to be confirmed with operating system
+            $ffiType = (PHP_OS === 'Linux') ? "long" : "int";
+
+            // declare cdata int / long types to pass by reference to position function
+            $top = FFI::new($ffiType);
+            $left = FFI::new($ffiType);
+            $bottom = FFI::new($ffiType);
+            $right = FFI::new($ffiType);
+
+            $this->ffi->mtGetBarStringPos($this->instance, $barcodeIndex, FFI::addr($left), FFI::addr($top), FFI::addr($right), FFI::addr($bottom));
+
+            // set referenced types
+            $barcodeScanResult->setTop($top->cdata);
+            $barcodeScanResult->setLeft($left->cdata);
+            $barcodeScanResult->setBottom($bottom->cdata);
+            $barcodeScanResult->setRight($right->cdata);
+
+            $barcodeScanResults[] = $barcodeScanResult;
+        }
+
+        return $barcodeScanResults;
+    }
+
+    /**
+     * Converts an FFI:CData<byteArray> to a readable string
+     *
+     * @param $byteArray
+     * @return string
+     */
+    private function ffiByteArrayToString($byteArray): string
+    {
+        $charType = FFI::type('char*');
+        $castedValue = FFI::cast($charType, $byteArray);
+        return FFI::string($castedValue);
+    }
+
+    /**
+     * will destroy the instance of the barcode tool kit
+     */
+    public function __destruct()
+    {
+        /* DestroyBarcodeInstance destroys an instance
+         * of the barcode toolkit and releases any resources used by the toolkit.
+         * This function only exists in the dll interface to the SDK.
```

Comment:
What about linux? Is there an alternative or is there no need to call a destroy method?

Reply:
Actually i cannot find any information about 'linux', only the instance while using the .dll library should be destroyed see here https://www.bardecode.com/en1/destroybarcodeinstance-destroy-an-instance-of-the-barcode-toolkit-win32-dll-only/

Update:
but in further researches i have found in die sdk a documentation where the destroy function is legit for all OS´s '

I take the os switch out , if you want to debug then the function should return 0 for success.
Want the Documentation ? Please DM me !

Reply:
If there is a method on the linux library (`.so`) we should use it 👍🏻